linting empty scaffolded chart results in panic

```
helm [master]$ bin/helm create mychart
Creating mychart
helm [master*]$ bin/helm lint mychart
panic: assignment to entry in nil map

goroutine 1 [running]:
panic(0x1169880, 0xc8202180f0)
	/usr/local/go/src/runtime/panic.go:464 +0x3e6
k8s.io/helm/pkg/chartutil.CoalesceValues(0xc820063e60, 0xc8201b5f80, 0xc8201a4a50, 0xc820515960, 0x0, 0x0)
	/Users/michellenoorali/gocode/src/k8s.io/helm/pkg/chartutil/values.go:153 +0x1f1
k8s.io/helm/pkg/lint/rules.Templates(0xc820515ac8)
	/Users/michellenoorali/gocode/src/k8s.io/helm/pkg/lint/rules/template.go:65 +0x542
k8s.io/helm/pkg/lint.All(0x7fff5fbffa06, 0x7, 0x0, 0x0, 0x0)
	/Users/michellenoorali/gocode/src/k8s.io/helm/pkg/lint/lint.go:35 +0x16a
main.lintCmd(0x21e0280, 0xc8202b1980, 0x1, 0x1, 0x0, 0x0)
	/Users/michellenoorali/gocode/src/k8s.io/helm/cmd/helm/lint.go:63 +0x18b
k8s.io/helm/vendor/github.com/spf13/cobra.(*Command).execute(0x21e0280, 0xc8202b1940, 0x1, 0x1, 0x0, 0x0)
	/Users/michellenoorali/gocode/src/k8s.io/helm/vendor/github.com/spf13/cobra/command.go:561 +0x62c
k8s.io/helm/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x21dfa80, 0x21e0280, 0x0, 0x0)
	/Users/michellenoorali/gocode/src/k8s.io/helm/vendor/github.com/spf13/cobra/command.go:651 +0x55c
k8s.io/helm/vendor/github.com/spf13/cobra.(*Command).Execute(0x21dfa80, 0x0, 0x0)
	/Users/michellenoorali/gocode/src/k8s.io/helm/vendor/github.com/spf13/cobra/command.go:610 +0x2d
main.main()
	/Users/michellenoorali/gocode/src/k8s.io/helm/cmd/helm/helm.go:89 +0x23
```

I believe this is due to empty values.yaml

--- a/pkg/chartutil/values.go
+++ b/pkg/chartutil/values.go
@@ -110,9 +110,9 @@ func tableLookup(v Values, simple string) (Values, error) {
 
 // ReadValues will parse YAML byte data into a Values.
 func ReadValues(data []byte) (vals Values, err error) {
-	vals = make(map[string]interface{})
-	if len(data) > 0 {
-		err = yaml.Unmarshal(data, &vals)
+	err = yaml.Unmarshal(data, &vals)
+	if len(vals) == 0 {
+		vals = Values{}
 	}
 	return
 }

--- a/pkg/chartutil/values_test.go
+++ b/pkg/chartutil/values_test.go
@@ -53,6 +53,18 @@ water:
 		t.Fatalf("Error parsing bytes: %s", err)
 	}
 	matchValues(t, data)
+
+	tests := []string{`poet: "Coleridge"`, "# Just a comment", ""}
+
+	for _, tt := range tests {
+		data, err = ReadValues([]byte(tt))
+		if err != nil {
+			t.Fatalf("Error parsing bytes: %s", err)
+		}
+		if data == nil {
+			t.Errorf(`YAML string "%s" gave a nil map`, tt)
+		}
+	}
 }
 
 func TestReadValuesFile(t *testing.T) {

